Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs

```diff
@@ -530,7 +530,12 @@ public ResultCode RecvFrom(ServiceCtx context)
 
                     receiveRegion.Dispose();
 
-                    context.Memory.Write(sockAddrOutPosition, BsdSockAddr.FromIPEndPoint(endPoint));
+                    var bsdSockAddr = BsdSockAddr.FromIPEndPoint(endPoint);
+
+                    if (sockAddrOutSize != 0 && sockAddrOutSize >= (ulong) bsdSockAddr.Size())
```

Comment:
From my understanding here you are only implementing BsdSockAddr.Size to get the size of a single element size, you can just use [Unsafe.SizeOf API](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.sizeof?view=net-7.0).
```suggestion
                    if (sockAddrOutSize != 0 && sockAddrOutSize >= (ulong) Unsafe.SizeOf<BsdSockAddr>())
```